code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.cacheImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.getAssetCacheLocation
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.getCachedImage
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.getCarouselIndices
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.populateFilmstripCarouselImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.populateManualCarouselImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.ManualCarouselPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedManualCarousalTemplate
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockkClass
+import io.mockk.mockkObject
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class ManualCarouselNotificationBuilderTest {
+
+    private lateinit var context: Context
+    private lateinit var pushTemplate: ManualCarouselPushTemplate
+    private lateinit var expandedLayout: RemoteViews
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+
+    @Before
+    fun setUp() {
+        context = RuntimeEnvironment.getApplication()
+        pushTemplate = provideMockedManualCarousalTemplate(false)
+        trackerActivityClass = mockkClass(Activity::class, relaxed = true).javaClass
+        broadcastReceiverClass = mockkClass(BroadcastReceiver::class, relaxed = true).javaClass
+        mockkObject(PushTemplateImageUtils)
+        mockkObject(BasicNotificationBuilder)
+        mockkObject(PendingIntentUtils)
+        expandedLayout = mockkClass(RemoteViews::class)
+    }
+
+    @Test
+    fun `construct returns NotificationCompat Builder for valid inputs`() {
+        val result = ManualCarouselNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertNotNull(result)
+    }
+
+    @Test
+    fun `construct returns BasicNotificationBuilder if download image count is less than 3`() {
+        every { cacheImages(any()) } answers { 2 }
+        ManualCarouselNotificationBuilder.construct(
```

Comment:
nit: can we capture the result and verify its non null similar to what was done in https://github.com/adobe/aepsdk-ui-android/pull/25/files#diff-9304ffe0e17f8b8dd4d1ecc50b92f9a23bca3d0de213b2c650114964f1d8e7b6R71 and https://github.com/adobe/aepsdk-ui-android/pull/25/files#diff-9304ffe0e17f8b8dd4d1ecc50b92f9a23bca3d0de213b2c650114964f1d8e7b6R77?

Reply:
@rymorale I've written that test just above this test to make sure that we're able to construct the builder successfully, whereas the objective of this test is to unit test the below if condition inside our constructor
`if (downloadedImagesCount < PushTemplateConstants.DefaultValues.CAROUSEL_MINIMUM_IMAGE_COUNT) {
            Log.warning(LOG_TAG, SELF_TAG, "Less than 3 images are available for the manual carousel push template, falling back to a basic push template.")
            return BasicNotificationBuilder.fallbackToBasicNotification(
                context,
                trackerActivityClass,
                broadcastReceiverClass,
                pushTemplate.data
            )
        }`
for fallback to basic push template notification builder, therefore in order verify is is falling back properly or not i've written this unit test